### PR TITLE
Pre-0.1 release

### PR DESCRIPTION
**Pull request by fplk**
May update to FfDL.

We also need to fix CI before merging.

Please test thoroughly before accepting. Need to set `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` now before `make test-submit`.

Just ran it successfully against IBM Cloud and a DIND Kubernetes cluster, cmp. https://github.ibm.com/Falk-Pollok/ffdl-staging/wiki/Running-FfDL-Staging

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: lcm/service/lcm/constants.go [hunk 1]
```diff
@@ -77,9 +79,9 @@ const (
 	loadModelMilliCPU=20
 	loadModelMemInMB=50
 	loadTrainingDataMilliCPU=20
-	loadTrainingDataMemInMB=300
-	logCollectorMilliCPU=60
-	logCollectorMemInMB=300
+	loadTrainingDataMemInMB=100
+	logCollectorMilliCPU=20
+	logCollectorMemInMB=100
```
Comment (Tomcli) [inline lcm/service/lcm/constants.go]: Log collector may have memory insufficiency with only 100MB memory. We may want to increase the constant for now or make it configurable using viper.

File: etc/examples/tf-summary-model/gpu-manifest.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ name: tf_mnist_with_summaries_gpu_tutorial
 description: Convolutional network model using tensorflow (GPU)
 version: "1.0"
 gpus: 1
-cpus: 2
```
Comment (Tomcli) [inline etc/examples/tf-summary-model/gpu-manifest.yml]: Please revert the changes for this example

**Comment by Tomcli** [comment]
Overall this PR works fine. However, the instructions are very specific for IBM Public Cloud and it's confusing for general users. Also, we need to change the object storage driver to the [open source version](https://github.com/IBM/ibmcloud-object-storage-plugin) before we can merge this PR.

**Comment by animeshsingh** [comment]
Open Issues:
1. Including the open source object storage plugin
2. Ironing out Minikube deployment (on Mac)
3. Usability and setup across multiple Kubernetes deployments

File: README.md [hunk 3]
```diff
@@ -12,8 +12,6 @@ FfDL is a collaboration platform for:
 
 ![ffdl-architecture](docs/images/ffdl-architecture.png)
 
-To know more about the architectural details, please read [here](design/design_docs.md)
-
```
Comment (Tomcli) [inline README.md]: Can you restore this 2 line?

File: docs/developer-guide.md [hunk 4]
```diff
@@ -16,12 +16,6 @@ Install:
 * `Go` is very specific about directory layouts. Make sure to set your `$GOPATH` and clone this repo to a directory
 `$GOPATH/src/github.com/IBM/FfDL` before proceeding with the next steps.
 
-> If you are developing on Minikube, please run the following commands to configure your Minikube and set up the Docker client to use the Minikube's Docker Daemon. Then start to build your own Docker images.
-> ```shell
-> export VM_TYPE=minikube
-> make minikube
-> eval $(minikube docker-env)
-> ```
```
Comment (Tomcli) [inline docs/developer-guide.md]: Can you revert these changes?

**Comment by Tomcli** [comment]
For Minikube, If we using the old `s3_datastore` mode (has bottleneck) instead of COS, then we don't need to have the daemonset and flex driver install on Minikube. However, it has a bug where the log data is not stored, so the GUI and CLI won't able to see the logs.

**Comment by fplk** [comment]
DO NOT MERGE, YET, PLEASE.

**Comment by animeshsingh** [comment]
I am going to go through the instructions step by step tomorrow. Pending any issues, will move for merge. @Tomcli can you confirm if DIND is flowing smoothly for MAC?

**Comment by sboagibm** [comment]
And if DIND is working smoothly for Mac OS, we need to change instructions from Minikube to DIND.  May be a bit more complex because of Docker and Kubernetes setup, but maybe we just point them somewhere.  -s
 
----- Original message -----From: Animesh Singh <notifications@github.com>To: IBM/FfDL <FfDL@noreply.github.com>Cc: sboagibm <scott_boag@us.ibm.com>, Review requested <review_requested@noreply.github.com>Subject: Re: [IBM/FfDL] Merge 20180514 1536 (#79)Date: Tue, Jun 19, 2018 11:41 AM 
I am going to go through the instructions step by step tomorrow. Pending any issues, will move for merge. @Tomcli can you confirm if DIND is flowing smoothly for MAC?

**Comment by Tomcli** [comment]
@fplk DIND works on MacOS, but we definitely need to rewrite the instructions and scripts to be capable with MacOS. We can discuss it more on tomorrow's call.

File: glide.yaml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package: github.com/IBM/FfDL
+package: github.com/IBM/FfDL/
```
Comment (ckadner) [inline glide.yaml]: the trailing slash character `/` seems to cause a warning:

```
[FfDL]$ glide install

[WARN]	The name listed in the config file (github.com/IBM/FfDL/) does not match the current location (github.com/IBM/FfDL)
[WARN]  ...
```

**Comment by fplk** [comment]
Do the recent changes work on DIND, yet? On Ubuntu 16.04 for Make target `quickstart-deploy` I get an error:
```
ffdlr@proteon:~/go/src/github.com/IBM/FfDL$ make quickstart-deploy
Deploying services to Kubernetes. This may take a while.
/bin/sh: 2: Syntax error: end of file unexpected (expecting "fi")
Makefile:100: recipe for target 'quickstart-deploy' failed
make: *** [quickstart-deploy] Error 2
```

**Comment by fplk** [comment]
OK, usability improvements work smoothly on DIND now. Great job, Tommy - thanks for the PR.

File: README.md [hunk 6]
```diff
@@ -62,47 +61,42 @@ To know more about the architectural details, please read [here](design/design_d
...
 
-To install FfDL to a proper IBM Cloud Kubernetes cluster, make sure `kubectl` points to the right namespace
-and your machine is logged in with `bx login`, then deploy the platform services:
-``` shell
-export VM_TYPE=ibmcloud
-export CLUSTER_NAME=<Your Cluster Name> # Replace <Your Cluster Name> with your IBM Cloud Cluster Name
-make deploy
+# Change the storage class to what's available on your Cloud Kubernetes Cluster.
+export SHARED_VOLUME_STORAGE_CLASS="ibmc-file-gold";
+make deploy-plugin
```
Comment (ckadner) [inline README.md]: are we missing a `helm init` before `make deploy-plugin` here? without that I get `Error: could not find a ready tiller pod`:
```
[ffdl]$ make deploy-plugin

Deploying services to Kubernetes. This may take a while.
Installing helm/tiller
Error: could not find a ready tiller pod
Wait while kubectl get pvc shows static-volume-1 in state Pending
Creating persistent volume claim 1
Error: could not find a ready tiller pod
persistentvolumeclaim "static-volume-1" created
Error from server (NotFound): configmaps "static-volumes" not found

Using volumes with label type=dlaas-static-volume:
NAME              STATUS    VOLUME    CAPACITY   ACCESS MODES   STORAGECLASS     AGE
static-volume-1   Pending                                       ibmc-file-gold   1s

configmap "static-volumes" created

```

File: README.md [hunk 7]
```diff
@@ -62,47 +61,42 @@ To know more about the architectural details, please read [here](design/design_d
...
+make quickstart-deploy
```
Comment (ckadner) [inline README.md]: I'm using the Mac OS Terminal App (Bash 3.2) and I get this error message: 
```
/bin/sh: @set: command not found
```
although seemingly without negative consequences. The `make` target succeeds.


```
$ make quickstart-deploy

collecting existing pods
calling big command
/bin/sh: @set: command not found
Deploying the stack via Helm. This will take a while.
...
STATUS: DEPLOYED
...
```

File: README.md (hunk 6 above)
Comment (Tomcli) [reply README.md]: Oh, sorry we did add helm init, but we forgot to put a sleep loop to wait for tiller to be ready.
